### ISSUE TITLE
refactor: make the pkg/oonimkall package more easily testable

### ISSUE THREAD

**Issue by bassosimone** [ooni/probe#1903]
I noticed this issue while working on https://github.com/ooni/probe-cli/pull/539. The `pkg/oonimkall` package is not easy to test and we're duplicating a bunch of tests. This happens because part of the package is internal. So, we're duplicating some effort in testing the same things twice. I would argue it's simpler and it would give more confidence if all the code was inside the same package rather than being split across two packages. I'm going to explore this possibility. If it seems it leads to an improvement, then I am going to move code from the internal package directly in `pkg/oonimkall`.

**Comment by bassosimone** [comment]
In https://github.com/ooni/probe-cli/commit/c4eb6826066c2449cddc84474f136430cccc91b5, I fixed a crash where we could post on a closed channel. I think that fix is good but there is opportunity to write slightly more correct code that does not depend on a timeout. Yesterday I did a bit of research with a toy repository to figure this out and it seems I've found a very small tweak (passing around a channel signalling close) that does the trick.
